prow/cmd/gangway/main.go

```diff
@@ -156,26 +157,31 @@ func main() {
 		}
 	}
 
-	gitClient, err := o.github.GitClientFactory(o.cookiefilePath, &o.config.InRepoConfigCacheDirBase, o.dryRun, false)
-	if err != nil {
-		logrus.WithError(err).Fatal("Error getting Git client.")
+	gw := gangway.Gangway{
+		ConfigAgent:       configAgent,
+		ProwJobClient:     prowjobClient,
 	}
 
-	cacheGetter, err := config.NewInRepoConfigCache(o.config.InRepoConfigCacheSize, configAgent, gitClient)
-	if err != nil {
-		logrus.WithError(err).Fatal("Error creating InRepoConfigCacheGetter.")
+	// InRepoConfig getter.
+	if o.config.MoonrakerAddress != "" {
+		gw.InRepoConfigGetter = moonraker.NewClient(o.config.MoonrakerAddress, 10*time.Second)
```

Comment:
There is no verification of the ability to connect with Moonraker provided by NewClient. The alternatives here Fatal if there are any issues with dependencies. I wonder if it makes sense to establish Moonraker is alive and we can connect, otherwise Fatal?

Reply:
I could add a basic `/ping` endpoint that returns 200 if the service is up.

Reply:
Hmm, actually, we do have a polling timeout pattern in one of the integration tests I wrote a while back. I think we can do the same thing here to give a little bit of slack if the service is not immediately available (presumably during startup when different services could be deployed in a different order, the exact ordering of which is outside our control and could be unique to Prow installations).

Reply:
Done.

Reply:
I think there are some kinks to sort out with the integration tests. Stay tuned...

Reply:
Looks like I had a typo in the path being served for the new "/ping" part. Fixed. Tests should be passing now (modulo flakes).